internal/incident/incident.go

```diff
@@ -65,7 +66,9 @@ func (i *Incident) HasManager() bool {
 }
 
 // ProcessEvent processes the given event for the current incident.
-func (i *Incident) ProcessEvent(ctx context.Context, tx *sqlx.Tx, ev event.Event, created bool) error {
+// If non-acknowledgement event is given, this returns pending incident notification histories of type *HistoryRow
+// and are used to notify the recipients based on these entries.
+func (i *Incident) ProcessEvent(ctx context.Context, tx *sqlx.Tx, ev event.Event, created bool) ([]*HistoryRow, error) {
```

Comment:
`HistoryRow` has lots of fields and it's a bit unclear, which ones are actually used. I think a dedicated type containing the very specific information needed would be nicer (so something like contact, channel, ..., and incident history entry ID to update with the result).